src/main/java/org/radarcns/security/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,103 @@
+package org.radarcns.security.filter;
+
+import org.apache.http.HttpHeaders;
+import org.radarcns.auth.authentication.TokenValidator;
+import org.radarcns.auth.config.ServerConfig;
+import org.radarcns.auth.config.YamlServerConfig;
+import org.radarcns.auth.exception.TokenValidationException;
+import org.radarcns.config.managementportal.config.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.lang.ref.SoftReference;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Locale;
+
+/**
+ * Created by dverbeec on 27/09/2017. Updated in Rest-Api by yatharthranjan on 10/11/2017.
+ */
+public class AuthenticationFilter implements Filter {
+
+    private static Logger log = LoggerFactory.getLogger(AuthenticationFilter.class);
+    public static final String TOKEN_ATTRIBUTE = "jwt";
+
+    private static SoftReference<TokenValidator> validator = new SoftReference<>(null);
```

Comment:
What was the reason for making it a SoftReference?

Reply:
In general, it allows a static variable to be garbage collected if necessary. But if this isn't going to happen anyway, it could be a direct static variable instead.